Source/NSTextFinder.m

```diff
@@ -0,0 +1,305 @@
+/* Implementation of class NSTextFinder
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 02-08-2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSArchiver.h>
+
+#import "AppKit/NSTextFinder.h"
+#import "AppKit/NSTextView.h"
+
+#import "GSTextFinder.h"
+
+@implementation NSTextFinder
+
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      // initialize finder...
+      _finder = [GSTextFinder sharedTextFinder];
+    }
+  return self;
+}
+
+// Operation methods
+- (void) _showFindInterface
+{
+  [_finder showFindPanel];
+}
+
+- (void) _nextMatch
+{
+  [_finder findStringInTextView: (NSTextView *)_client
+                        forward: YES];
+}
+
+- (void) _previousMatch
+{
+  [_finder findStringInTextView: (NSTextView *)_client
+                        forward: NO];
+}
+
+- (void) _replaceAll
+{
+  [_finder replaceAllInTextView: (NSTextView *)_client
+                onlyInSelection: NO];
+}
+
+- (void) _replace
+{
+  [_finder replaceStringInTextView: (NSTextView *)_client];
+}
+
+- (void) _replaceAndFind
+{
+   [_finder replaceStringInTextView: (NSTextView *)_client];
+   [_finder findStringInTextView: (NSTextView *)_client
+                         forward: YES];
+}
+
+- (void) _setSearchString
+{
+  [_finder takeFindStringFromTextView: (NSTextView *)_client];
+}
+
+- (void) _replaceAllInSelection
+{
+  [_finder replaceAllInTextView: (NSTextView *)_client
+                onlyInSelection: YES];
+}
+
+- (void) _selectAll
+{
+}
+
+- (void) _selectAllInSelection
+{
+}
+
+- (void) _hideFindInterface
+{
+}
+
+- (void) _showReplaceInterface
+{
+  [_finder showFindPanel];
+}
+
+- (void) _hideReplaceInterface
+{
+}
+
+// Validating and performing
+- (void) performAction: (NSTextFinderAction)op
```

Comment:
This class needs also to implement the NSResponder method performTextFinderAction: (and for compatibly with old versions also performFindPanelAction:) which gets the tag of the sender and calls this method here.

Reply:
Done.